### ISSUE TITLE
inconsistant values

### ISSUE THREAD

**Issue by kilabyte** [bhaskar-anil429/MCP9600#2]
Using the lib i am able to read the temperatures but i keep getting inconsistent values. I should be getting somwhere between 3-6 degrees C but i get this.

<img width="230" alt="Screen Shot 2020-12-22 at 3 47 05 PM" src="https://user-images.githubusercontent.com/487948/102943495-16d34280-446d-11eb-8367-a9213aab7853.png">

any ideas? It is a K type thermocouple

**Comment by bhaskar-anil429** [comment]
what values do you get if no thermocouple is connected ?
can you share picture of the setup or sch if you have?

**Comment by kilabyte** [comment]
they are the same. here is the setup
![3657717d-04c2-4f78-ba6c-75769c80b799](https://user-images.githubusercontent.com/487948/102945080-982cd400-4471-11eb-9fbb-3ecdafb86072.jpeg)

**Comment by bhaskar-anil429** [comment]
did you try connecting the MCP9600 directly to PI ( ship mux)

**Comment by kilabyte** [comment]
yup exactly the same issue. Even tried changed the address on the board for the mcp. Tried different mcp board, different pi, different mux. Sometimes itll work, sometimes itll be every other reading, and most of the time itll just be like the above.

**Comment by kilabyte** [comment]
I also loaded up this library (https://github.com/pimoroni/mcp9600-python) and if i call the "cold" function it seems to be more consistent but thats the ambient temp. If i call "hot" function then i get the same inconsistencies as above for the temp on the thermocouple

**Comment by kilabyte** [comment]
any update?

**Comment by kilabyte** [comment]
turns out it is the use of the raspi4 makes it do this. if i run this on a raspi3 then everything runs perfectly
